src/model/mod.rs

```diff
@@ -48,8 +50,145 @@ pub enum PlayingItem {
     Episode(show::FullEpisode),
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Id<'id> {
+    _type: Type,
+    id: &'id str,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Display, Error)]
+pub enum IdError {
+    InvalidPrefix,
+    InvalidFormat,
+    InvalidType,
+    InvalidId,
+}
+
+impl std::fmt::Display for Id<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.write_str("spotify:")?;
+        f.write_str(self._type.as_ref())?;
+        f.write_str(":")?;
+        f.write_str(self.id)?;
+        Ok(())
+    }
+}
+
+impl AsRef<str> for Id<'_> {
+    fn as_ref(&self) -> &str {
+        self.id
+    }
+}
+
+impl Id<'_> {
```

Comment:
This type should have more documentation regarding what's considered an ID and what's an URI. It should also note that URIs may use `/` or `:` as separators, and that IDs are made up of alphanumeric characters only.

Reply:
I see. So you want to expose the `Id` type to the user and make it a part of public API? This first proposal was intended as part of an internal API. If you think it's better to make it public, then I agree, we need:

1. Good documentation on `Id` type
2. Make all methods accept `Id` instead of `&str` when Spotify id or URI is required

Reply:
Yes, sorry if I didn't make my idea clearer before. Having `Id` as public would give the user more options on these kinds of parameters, and makes use of the type safety Rust provides.

Why not use `String` rather than `Cow<&str>`? That's how these structs usually go.

Very thorough documentation, by the way! I love it.

Reply:
Ah okay I just read in another comment that what you want with this is to avoid allocations. I'm not that experienced with that and I'm not sure at what point it's worth just not using `String` at all. I'd have to read more about it.